### PR TITLE
Update index.md

### PR DESCRIPTION
**Pull request by jeroen-van-dijk**
AuthorizationStatus seems to be nested inside FirebaseMessagingTypes

### Checklist

- I read the [Contributor Guide](/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [ ] Yes
- My change supports the following platforms;
  - [ ] `Android`
  - [ ] `iOS`
- My change includes tests;
  - [ ] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [ ] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [ ] No

### Test Plan

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

**Comment by mikehardy** [comment]
@jeroen-van-dijk needs the CLA handled before can merge

**Comment by Salakar** [comment]
Hmm actually I think this should just be accessed via `messaging.AuthorizationStatus` rather than importing anything specific

File: docs/messaging/usage/index.md [hunk 1]
```diff
@@ -49,12 +49,12 @@ iOS prevents messages containing notification (or 'alert') payloads from being d
 This module provides a `requestPermission` method which triggers a native permission dialog requesting the user's permission:
 
 ```js
-import messaging, { AuthorizationStatus } from '@react-native-firebase/messaging';
+import messaging, { FirebaseMessagingTypes } from '@react-native-firebase/messaging';
```
Comment (Salakar) [inline docs/messaging/usage/index.md:52]:
```suggestion
import messaging from '@react-native-firebase/messaging';
```

File: docs/messaging/usage/index.md [hunk 2]
```diff
@@ -49,12 +49,12 @@ iOS prevents messages containing notification (or 'alert') payloads from being d
...
 
 async function requestUserPermission() {
   const authStatus = await messaging().requestPermission();
   const enabled =
-    authStatus === AuthorizationStatus.AUTHORIZED || authStatus === AuthorizationStatus.PROVISIONAL;
+    authStatus === FirebaseMessagingTypes.AuthorizationStatus.AUTHORIZED || authStatus === FirebaseMessagingTypes.AuthorizationStatus.PROVISIONAL;
```
Comment (Salakar) [inline docs/messaging/usage/index.md:57]:
```suggestion
    authStatus === messaging.AuthorizationStatus.AUTHORIZED || authStatus === messaging.AuthorizationStatus.PROVISIONAL;
```

**Comment by mikehardy** [comment]
Obsoleted by #3868 which has CLA signed

**Pull request closed without merging**